src/site/content/en/blog/form-fatigue-validate-user-input/index.md

```diff
@@ -0,0 +1,308 @@
+  // case 2 - pattern mismatch
+  else if (emailInput.validity.patternMismatch)  
+    emailInput.setCustomValidity('Your email address isn\'t valid (e.g. name@gmail.com)');  
+  // ...   
+} else {  
+  emailInput.setCustomValidity('');  
+}  
+```
+
+#### Case 1: Missing required email address
+
+<figure class="w-figure">
+  <img src="image-missing-field-default.png" alt="default validation message for missing field" style="max-width: 400px;">
+</figure>
+
+<!-- TODO (mdiblasio): update image / message, label should be above -->
+
+{% Compare 'worse', 'Less helpful' %}
+Default validation message is vague.
+{% endCompare %}
+
+<figure class="w-figure">
+  <img src="image-missing-field-custom.png" alt="custom validation message for missing field" style="max-width: 400px;">
+</figure>
+
+<!-- TODO (mdiblasio): update image / message, label should be above -->
+
+{% Compare 'better', 'More helpful' %}
+Custom validation message is specific.
+{% endCompare %}
+
+#### Case 2: Invalid email address
+
+<figure class="w-figure">
+  <img src="image-invalid-field-default.png" alt="default validation message for invalid field" style="max-width: 400px;">
+</figure>
+
+<!-- TODO (mdiblasio): update image / message, label should be above -->
+
+{% Compare 'worse', 'Less helpful' %}
+Default validation message is technincal and may be difficult for some users to understand.
+{% endCompare %}
+
+<figure class="w-figure">
+  <img src="image-invalid-field-custom.png" alt="custom validation message for invalid field" style="max-width: 400px;">
+</figure>
+
+<!-- TODO (mdiblasio): update image / message, label should be above -->
+
+{% Compare 'better', 'More helpful' %}
+Custom validation message is  non-technical and easy to understand.
+{% endCompare %}
+
+
+### Show feedback in real-time
+
+<!-- <div class="w-stats">
+  <div class="w-stat">
+    <p class="w-stat__figure">46<sub class="w-stat__sub">%</sub></p>
+    <p class="w-stat__desc"><a href="https://bit.ly/msites-benchmark-uk-2018">Did not implement error messaging in real-time</a></p>
+  </div>
+</div> -->
+
+
+You should check form data as users complete each field, rather than waiting for them to submit. This makes it quicker for users to correct mistakes, and makes it less likely that invalid or insecure data will be submitted.
+
++   **Use CSS pseudo-classes** `invalid` and `valid` to apply custom styling:
+
+
+```css
+input:not(:focus):invalid {  
+  background-color: #FFD9D9;  
+}  
+input:not(:focus):valid {  
+  background-color: #D9FFD9;  
+}  
+```  
+<!-- [https://codepen.io/samdutton/pen/ZPrpmL](https://codepen.io/samdutton/pen/ZPrpmL) -->
+
+<figure class="w-figure">
+  <img src="https://screenshot.googleplex.com/noq2iTxzuh1.png" alt="CSS validation styling" style="max-width: 400px;">
+  <figcaption class="w-figcaption">
+    Example - CSS validation styling.
+  </figcaption>
+</figure>
+
+<div class="w-aside w-aside--caution">
+  Do not rely on color alone to communicate status. Colors
```

Comment:
May want to mention this is for accessibility reasons to benefit users with color vision impairments.